src/lammpsio/snapshot.py

```diff
@@ -164,31 +219,114 @@ def to_hoomd_gsd(self, type_map=None):
                 for typeidx, typeid in enumerate(sorted_typeids):
                     frame.particles.typeid[self.typeid == typeid] = typeidx
             else:
-                frame.particles.types = list(type_map.values())
-                reverse_type_map = {
-                    typeid: typeidx for typeidx, typeid in enumerate(type_map.keys())
-                }
-                for i, typeid in enumerate(self.typeid):
-                    frame.particles.typeid[i] = reverse_type_map[typeid]
+                if isinstance(type_map, dict):
+                    self.label = LabelMap(map=type_map)
+                    warnings.warn(
+                        "type_map should be a TypeMap object.",
+                        DeprecationWarning,
+                        stacklevel=2,
+                    )
+                frame.particles.types = self.label.types
+                Snapshot._set_type_id(
+                    self,
+                    self.typeid,
+                    frame.particles.typeid,
+                    self.label,
+                )
         if self.has_charge():
             frame.particles.charge = self.charge.copy()
         if self.has_mass():
             frame.particles.mass = self.mass.copy()
         if self.has_molecule():
             frame.particles.body = self.molecule - 1
+        if self.has_bonds():
+            frame.bonds.N = self.bonds.N
+            frame.bonds.types = self.bonds.id
+            frame.bonds.group = self.bonds.members - 1
+            if self.bonds.has_typeid():
+                frame.bonds.typeid = numpy.zeros(self.bonds.N, dtype=int)
+                if self.bonds.label is not None:
+                    frame.bonds.types = self.bonds.label.types
+                    Snapshot._set_type_id(
+                        self,
+                        self.bonds.typeid,
+                        frame.bonds.typeid,
+                        self.bonds.label,
+                    )
+                else:
+                    sorted_typeids = numpy.sort(numpy.unique(self.bonds.typeid))
+                    frame.bonds.types = [str(typeid) for typeid in sorted_typeids]
```

Comment:
I think it would be clearer & reduce duplication to generate a `LabelMap` from these lines, then just use `_set_type_id`.